### PR TITLE
Molden Writer update

### PR DESCRIPTION
**Pull request by sgoodlett**
## Description
Moved old Molden writer to Python, fixed normalization issue, and added test cases

## Checklist
- [X] Tests added for any new features
- [X] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [ ] Ready for merge

### CODE REVIEW

**Comment by JonathonMisiewicz** [comment]
Happy to see this up!  Go ahead and mark this ready for review. You want to mark with a single "X" rather than "Y " or " Y" so you get pretty checkmarks.

This PR will close #1973. I was pretty heavily involved in this, so I _probably_ shouldn't review this one.

File: tests/PEDRA.OUT__19028 [hunk 1]
```diff
@@ -0,0 +1,247 @@
+Point group: C1
```
Comment (JonathonMisiewicz) [inline tests/PEDRA.OUT__19028:1]: This file shouldn't be here.

File: psi4/src/psi4/libmints/writer.h [hunk 2]
```diff
@@ -81,6 +81,22 @@ class PSI_API MoldenWriter {
                std::shared_ptr<Vector> OccB, bool dovirtual);
 };
 
+class PSI_API MoldenWriter {
```
Comment (JonathonMisiewicz) [inline psi4/src/psi4/libmints/writer.h:84]: I think you misunderstood me earlier. Get rid of this "new" class and move the `PSI_DEPRECATED` warnings to the functions of the _existing_ `MoldenWriter` class. This is probably why the build is failing.

**Review by jturney — APPROVED** [review]

**Comment by loriab** [comment]
Just checking that the extra precision added https://github.com/psi4/psi4/pull/1510/files didn't get lost.

**Comment by JonathonMisiewicz** [comment]
> Just checking that the extra precision added https://github.com/psi4/psi4/pull/1510/files didn't get lost.

We studied numerical precision and couldn't justify going out to the **17** decimal places reported after #1510: the Python code couldn't reproduce the C++ code out that far. My memory is hazy, but I recall some errors that were normally in the 14th decimal place but could rarely reach to the 12th decimal place, which I suspected were related to the numerical precision of S^(-1/2). This made me want to cut back the number of decimal places to something less fragile.

I [checked in](https://github.com/psi4/psi4/issues/1973) about why we had 17 decimal places in the first place, and the person who made the change simply assumed that we had that many decimals of precision. In my judgment, we do not, and it was a mistake to ever report that many decimal places.

I _also_ checked that 10 decimal places was enough to keep #1486 fixed. (That was the issue that originally prompted the increased precision in our Molden files.) While the 4 decimal places of the pre-#1486 era had the bug, the 10 decimal places of this PR did not.

While we could _maybe_ go out to 11 or 12 decimal places, trying to get 17 decimal places of precision would require an analysis of round-off error in every line of the SCF code. This seems far beyond the scope of the PR.

File: psi4/src/psi4/libmints/writer.cc [hunk 3]
```diff
@@ -46,6 +46,12 @@
 using namespace psi;
 ;
 
+
+MoldenWriter::MoldenWriter(std::shared_ptr<Wavefunction> wavefunction) : wavefunction_(wavefunction) {
```
Comment (jturney) [inline psi4/src/psi4/libmints/writer.cc:50]: Why did you duplicate the function from line 55. The compiler is reporting an error.  This should have come up on your local compile. One of them must be deleted.

**Review by jturney — CHANGES_REQUESTED** [review]

**Comment by JonathonMisiewicz** [comment]
We might as well fix #2238 while we're at it. The problematic line of code is [right here](https://github.com/psi4/psi4/pull/2229/files#diff-21e554583cfa696eeabb7d40e85e87689e1bd9f72ff88e593fdac893c6b61ac9R356).

**Review by JonathonMisiewicz — APPROVED** [review]
That's much better, thank you. I'll approve this in the interests of expediting 1.4. If anybody thinks we need three _independent_ reviewers, just say so.

**Review by jturney — APPROVED** [review]

**Review by loriab — APPROVED** [review]

**Review by andysim — APPROVED** [review]
Excellent work!  This is exactly the kind of thing the belongs py-side.  Kudos for the extensive tests.

File: psi4/driver/p4util/writer.py [hunk 4]
```diff
@@ -198,3 +198,203 @@ def write_C_matrix(C, count):
...
+        NO_Ra = core.Matrix("NO Alpha Rotation Matrix", nmopi, nmopi)
+        occupation_a = core.Vector(nmopi)
+        MO_Da.diagonalize(NO_Ra, occupation_a, core.DiagonalizeOrder.Descending)
+        Ca = core.doublet(self.Ca(), NO_Ra, False, False)
+        epsilon_a = occupation_a
+        # Betas
+        #MO_Db = core.Matrix("MO Beta Density Matrix", nmopi, nmopi)
+        #MO_Db.transform(self.Db(), self.Cb().transpose())
+        MO_Db = self.Db_subset("MO")
+        MO_Db.print_out()
```
[the remark this answers is not in the document]
Comment (JonathonMisiewicz) [reply psi4/driver/p4util/writer.py:294]: Good catch. This is probably a debugging statement that survived.

**Pull request merged**